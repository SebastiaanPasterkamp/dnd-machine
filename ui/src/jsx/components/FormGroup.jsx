import React from 'react';
import LazyComponent from '../components/LazyComponent.jsx';

import _ from 'lodash';

export class FormGroup extends LazyComponent
{
    renderLabel(index, label) {
        return <span
                key={"label-" + index}
                className="nice-input-addon">
            {label}
        </span>;
    }

    render() {
        return <div className="nice-form-group">
            <label>{this.props.label}</label>
            {this.props.children}
        </div>;
    }
}

export default FormGroup;