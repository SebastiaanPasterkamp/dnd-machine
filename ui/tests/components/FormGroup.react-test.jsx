import React from 'react';
import FormGroup from 'components/FormGroup.jsx';
import renderer from 'react-test-renderer';

describe('Component: FormGroup', () => {
    it('should render', () => {
        const tree = renderer.create(
            <FormGroup label="test">
                Foo bar
            </FormGroup>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
