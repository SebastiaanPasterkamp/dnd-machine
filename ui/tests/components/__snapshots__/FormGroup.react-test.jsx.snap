// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: FormGroup should render 1`] = `
<div
  className="nice-form-group"
>
  <label>
    test
  </label>
  Foo bar
</div>
`;
